$OpenBSD: patch-cachessess_t_c,v 1.1 2021/12/04 12:06:14 tb Exp $

Skip tests that won't compile with opaque structures in libcrypto

Index: cachessess.t.c
--- cachessess.t.c.orig
+++ cachessess.t.c
@@ -130,7 +130,7 @@ START_TEST(cache_ssess_03)
 }
 END_TEST
 
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x3040000fL)
 START_TEST(cache_ssess_04)
 {
 	SSL_SESSION *s1, *s2;
@@ -174,7 +174,7 @@ cachessess_suite(void)
 	tcase_add_test(tc, cache_ssess_01);
 	tcase_add_test(tc, cache_ssess_02);
 	tcase_add_test(tc, cache_ssess_03);
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x3040000fL)
 	tcase_add_test(tc, cache_ssess_04);
 #endif
 	suite_add_tcase(s, tc);
